Python/compile.c

```diff
@@ -5380,6 +5493,38 @@ compute_code_flags(struct compiler *c)
     return flags;
 }
 
+// Merge *tuple* with constant cache.
+// Unlike merge_consts_recursive(), this function doesn't work recursively.
+static int
+merge_const_tuple(struct compiler *c, PyObject **tuple)
+{
+    Py_ssize_t len = PyTuple_GET_SIZE(*tuple);
+    if (len == 0) {
```

Comment:
I'm not 100% sure that all empty tuples are the tuple singleton. Would it hurt to double check that we already have the singleton by removing this special case?